File: contests/lc/1266.MinimumTimeVisitingAllPoints/solution.py
from typing import List


class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        if not points:
            return 0
        total = 0
        for i in range(1, len(points)):
            dx = abs(points[i][0] - points[i - 1][0])
            dy = abs(points[i][1] - points[i - 1][1])
            total += max(dx, dy)
        return total

File: contests/lc/1266.MinimumTimeVisitingAllPoints/solution.go
package solution

func minTimeToVisitAllPoints(points [][]int) int {
	if len(points) == 0 {
		return 0
	}
	total := 0
	for i := 1; i < len(points); i++ {
		dx := abs(points[i][0] - points[i-1][0])
		dy := abs(points[i][1] - points[i-1][1])
		if dx > dy {
			total += dx
		} else {
			total += dy
		}
	}
	return total
}

func abs(v int) int {
	if v < 0 {
		return -v
	}
	return v
}
